### PR TITLE
Update to Dear ImGUI 1.86

### PR DESCRIPTION
**Pull request by marsbar-hy**
Dear ImGUI:
- 100% drop-in replacement. However, TheForge DOES modify Dear ImGUI. Fortunately, there is a different way to make this compatible (see change notes below).

TheForge UI Wrapper:
- Modified SetDefaultStyle() to use Dear ImGUI types instead of engine type.
- Added Dear ImGUI defines for type conversions of Vec2, Vec4, etc. to engine types so that we don't have to modify Dear ImGUI directly.
- Commented out DestroyDemoWindow(), could not find the changelog for it

### CODE REVIEW

File: Common_3/OS/Interfaces/IUI.h [hunk 1]
```diff
@@ -44,6 +44,28 @@
...
 #define MAX_TITLE_STR_LENGTH 128
 
+/****************************************************************************/
+// HYPIXELMOD
+// MARK: - Dear imGUI implicit casts
+/****************************************************************************/
+#define IM_VEC2_CLASS_EXTRA \
+	ImVec2(const float2& f) \
+	{                       \
+		x = f.x;            \
```
Comment (marsbar-hy) [inline Common_3/OS/Interfaces/IUI.h:54]: The formatter went crazy, I will fix.

File: Common_3/OS/Interfaces/IUI.h [hunk 2]
```diff
@@ -44,6 +44,28 @@
 #define MAX_FORMAT_STR_LENGTH 30
 #define MAX_TITLE_STR_LENGTH 128
 
+/****************************************************************************/
+// HYPIXELMOD
```
Comment (NicolasHypixel) [inline Common_3/OS/Interfaces/IUI.h:48]: Maybe explain here shortly the intent

File: Common_3/OS/UI/UI.cpp [hunk 3]
```diff
@@ -2217,7 +2216,7 @@ void platformExitUserInterface()
 	}
 	pUserInterface->mComponents.clear();
 
-	ImGui::DestroyDemoWindow();
+	//ImGui::DestroyDemoWindow();
```
Comment (NicolasHypixel) [inline Common_3/OS/UI/UI.cpp:2219]: Please, remove commented code

File: Common_3/OS/Interfaces/IUI.h [hunk 4]
```diff
@@ -49,22 +49,22 @@
 // MARK: - Dear imGUI implicit casts
 /****************************************************************************/
 #define IM_VEC2_CLASS_EXTRA \
```
Comment (NicolasHypixel) [inline Common_3/OS/Interfaces/IUI.h:51]: This commit could be fixed-up with previous one.

**Review by NicolasHypixel — APPROVED** [review]
A few minor remarks.

File: Common_3/OS/Interfaces/IUI.h (hunk 2 above)
Comment (marsbar-hy) [reply Common_3/OS/Interfaces/IUI.h:48]: This is a token in the file indicating that we've modified an external library so next time we have to update it is easy to find.

**Pull request merged**